islandora_compound_object.module

```diff
@@ -261,6 +261,33 @@ function islandora_compound_object_islandora_compoundCModel_islandora_view_objec
 }
 
 /**
+ * Implements hook_page_build().
+ */
+function islandora_compound_object_page_build(&$page) {
```

Comment:
Did you try adding your logic to the above `islandora_compound_object_islandora_compoundCModel_islandora_view_object()` function?

I'm wondering if adding your check and redirect there wouldn't make more sense. I have to test, but I think this way it will run both functions which means 2 calls to `islandora_compound_object_get_parts()`. For performance reasons I think it would be beneficial to combine them.

I'll try and test this tomorrow.

Reply:
I use `hook_page_build()` because this hook is called early and not much unneeded work is done when the redirect is actually needed. I've also looked at `hook_boot()` (but that is too early) and `hook_init()` (but this is not called on cached pages). Also, project conditional_redirect uses the same hook.
Is `islandora_compound_object_islandora_compoundCModel_islandora_view_object()` not only called voor compound Cmodel objects? Will this work if "Only allow compound objects to have child objects associated with them" is disabled?
For performance reasons, isn't it better to implement a (single page request) cache in `islandora_compound_object_get_parts()` if it is expensive and it might be called multiple times? (isn't it called twice when JAIL is enabled?)